// curls/dist/cjs/Box/defaultTheme.js
'use strict'

exports.__esModule = true
exports.getBoxShadow = exports.borderRadiusScale = exports.borderWidthScale = exports.posUnit = exports.borderWidthUnit = exports.borderRadiusUnit = void 0

var _core = require('@emotion/core')

var _ref = {
  name: 'gokeqi',
  styles: 'box-shadow:none;',
}

const borderRadiusUnit = 'rem',
  borderWidthUnit = 'px',
  posUnit = 'px',
  borderWidthScale = [0, 1, 2, 4, 6, 10],
  borderRadiusScale = [0, 1 / 8, 1 / 4, 1 / 2, 1, 2, 1000],
  getBoxShadow = dp => {
    if (dp === void 0 || dp === false || dp === null || dp == 0) {
      return _ref
    }

    dp = parseInt(dp)
    const offset = dp / 16
    const blur = Math.abs(dp) / 16
    return (
      /*#__PURE__*/
      (0, _core.css)(
        'box-shadow:0 0 ',
        blur,
        'rem rgba(67,90,111,0.08),0 ',
        offset,
        'rem ',
        blur,
        'rem rgba(67,90,111,0.13);'
      )
    )
  }

exports.getBoxShadow = getBoxShadow
exports.borderRadiusScale = borderRadiusScale
exports.borderWidthScale = borderWidthScale
exports.posUnit = posUnit
exports.borderWidthUnit = borderWidthUnit
exports.borderRadiusUnit = borderRadiusUnit
