// curls/dist/cjs/Box/propTypes.js
'use strict'

exports.__esModule = true
exports.default = void 0

var _PropTypes = require('../PropTypes')

var _default = {
  // background-color
  bg: _PropTypes.boolOrString,
  // border radius
  br: _PropTypes.implicitNum,
  // border width
  bw: _PropTypes.implicitNum,
  // border color
  bc: _PropTypes.boolOrString,
  // box-shadow
  sh: _PropTypes.implicitNum,
  // margin
  m: _PropTypes.implicitNum,
  // padding
  p: _PropTypes.implicitNum,
  // position
  pos: _PropTypes.boolOrString,
  // display
  d: _PropTypes.boolOrString,
  // overflow
  ov: _PropTypes.boolOrString,
  // z-index
  z: _PropTypes.implicitNum,
  // width: __
  w: _PropTypes.implicitNum,
  // height: __
  h: _PropTypes.implicitNum,
  // top
  t: _PropTypes.implicitNum,
  // right
  r: _PropTypes.implicitNum,
  // bottmo
  b: _PropTypes.implicitNum,
  // left
  l: _PropTypes.implicitNum,
  // min-width
  minW: _PropTypes.implicitNum,
  // min-height
  minH: _PropTypes.implicitNum,
  // max-width
  maxW: _PropTypes.implicitNum,
  // max-height
  maxH: _PropTypes.implicitNum,
}
exports.default = _default
