// curls/dist/cjs/Flex/propTypes.js
'use strict'

exports.__esModule = true
exports.default = void 0

var _PropTypes = require('../PropTypes')

var _default = {
  // flex (display: flex)
  flex: _PropTypes.boolOrString,
  // flex--fixed (flex(0, 0, auto))
  fixed: _PropTypes.boolOrString,
  // flex--fluid (flex(1, 1, auto))
  fluid: _PropTypes.boolOrString,
  // flex--grow
  grow: _PropTypes.boolOrNum,
  // flex--shrink
  shrink: _PropTypes.boolOrNum,
  // flex-basis
  basis: _PropTypes.implicitNum,
  // flex--x
  row: _PropTypes.boolOrString,
  // flex--y
  column: _PropTypes.boolOrString,
  // flex--wrap
  wrap: _PropTypes.boolOrString,
  // flex--x-{left|center|right|around|between}
  justify: _PropTypes.boolOrString,
  // flex--y-{top|center|bottom|around|between|stretch}
  align: _PropTypes.boolOrString,
  // order: _
  order: _PropTypes.implicitNum,
  // flex--content-{top|center|bottom|around}
  alignContent: _PropTypes.boolOrString,
  // flex--self-{top|center|bottom|around}
  alignSelf: _PropTypes.boolOrString,
}
exports.default = _default
