// curls/dist/cjs/Divider.js
'use strict'

exports.__esModule = true
exports.Divider = exports.useDivider = void 0

var _core = require('@emotion/core')

var _core2 = require('@style-hooks/core')

var _object = _interopRequireDefault(require('empty/object'))

var _createComponent = _interopRequireDefault(require('./createComponent'))

var _Box = require('./Box')

var _utils = require('./utils')

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {default: obj}
}

function _extends() {
  _extends =
    Object.assign ||
    function(target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i]
        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key]
          }
        }
      }
      return target
    }
  return _extends.apply(this, arguments)
}

const defaultStyles = {
  name: '1vb2cio',
  styles: 'width:100%;min-height:1px;clear:both;position:relative;',
}

const useDivider = props =>
    (0, _core2.useStyles)(
      'divider',
      _object.default,
      (0, _utils.pushCss)(
        _extends(
          {
            role: 'separator',
            bg: 'translucentLight',
          },
          props
        ),
        defaultStyles
      )
    ),
  Divider = (0, _createComponent.default)('div', props =>
    (0, _Box.useBasicBox)(useDivider(props))
  )

exports.Divider = Divider
exports.useDivider = useDivider

if (process.env.NODE_ENV !== 'production') {
  const boxPropTypes = require('./Box/propTypes').default

  Divider.displayName = 'Divider'
  Divider.propTypes = boxPropTypes
}
