// curls/dist/cjs/index.js
'use strict'

exports.__esModule = true
var _exportNames = {
  MAX_Z_INDEX: true,
  assignDefaults: true,
  colorize: true,
  directionalScale: true,
  getDefault: true,
  getBreakpointOrder: true,
  getHoverQuery: true,
  loadImage: true,
  memoValue: true,
  memoThemeValue: true,
  nullIfFalsy: true,
  portalize: true,
  pushCss: true,
  supportsCSS: true,
  unit: true,
  ThemeProvider: true,
  defaultTheme: true,
  defaultColors: true,
  CurlsContext: true,
  CurlsConsumer: true,
  useCurls: true,
  Avatar: true,
  useAvatar: true,
  Box: true,
  useBox: true,
  useBasicBox: true,
  BasicBox: true,
  Breakpoint: true,
  useBreakpoint: true,
  Button: true,
  useButton: true,
  Card: true,
  useCard: true,
  Divider: true,
  useDivider: true,
  Drawer: true,
  DrawerBox: true,
  DrawerToggle: true,
  useDrawerBox: true,
  useDrawerContext: true,
  DrawerContext: true,
  DrawerConsumer: true,
  Drop: true,
  useDrop: true,
  Fade: true,
  useFade: true,
  FlexGrid: true,
  useFlexGrid: true,
  Flex: true,
  useFlex: true,
  Grid: true,
  useGrid: true,
  GridItem: true,
  useGridItem: true,
  Input: true,
  useInput: true,
  Link: true,
  useLink: true,
  A: true,
  NavLink: true,
  Modal: true,
  ModalBox: true,
  ModalToggle: true,
  useModalBox: true,
  useModalContext: true,
  ModalContext: true,
  ModalConsumer: true,
  Popover: true,
  PopoverMe: true,
  PopoverBox: true,
  usePopoverBox: true,
  usePopoverContext: true,
  PopoverContext: true,
  PopoverConsumer: true,
  Slide: true,
  useSlide: true,
  Text: true,
  useText: true,
  prettyText: true,
  TextArea: true,
  useTextArea: true,
  Transitionable: true,
  useTransitionable: true,
  useAriaPopup: true,
  useAriaPopupToggle: true,
  getDelay: true,
  css: true,
  jsx: true,
  withEmotionCache: true,
  CacheProvider: true,
  Global: true,
  keyframes: true,
  ClassNames: true,
  browserResets: true,
  containmentAttrs: true,
  cloneElement: true,
  createComponent: true,
  createRenderProp: true,
  useBreakpointValueParser: true,
  useToggleVisibility: true,
  useMediaQuery: true,
}
exports.ClassNames = exports.keyframes = exports.Global = exports.CacheProvider = exports.withEmotionCache = exports.jsx = exports.css = exports.getDelay = exports.useAriaPopupToggle = exports.useAriaPopup = exports.useTransitionable = exports.Transitionable = exports.useTextArea = exports.TextArea = exports.prettyText = exports.useText = exports.Text = exports.useSlide = exports.Slide = exports.PopoverConsumer = exports.PopoverContext = exports.usePopoverContext = exports.usePopoverBox = exports.PopoverBox = exports.PopoverMe = exports.Popover = exports.ModalConsumer = exports.ModalContext = exports.useModalContext = exports.useModalBox = exports.ModalToggle = exports.ModalBox = exports.Modal = exports.NavLink = exports.A = exports.useLink = exports.Link = exports.useInput = exports.Input = exports.useGridItem = exports.GridItem = exports.useGrid = exports.Grid = exports.useFlex = exports.Flex = exports.useFlexGrid = exports.FlexGrid = exports.useFade = exports.Fade = exports.useDrop = exports.Drop = exports.DrawerConsumer = exports.DrawerContext = exports.useDrawerContext = exports.useDrawerBox = exports.DrawerToggle = exports.DrawerBox = exports.Drawer = exports.useDivider = exports.Divider = exports.useCard = exports.Card = exports.useButton = exports.Button = exports.useBreakpoint = exports.Breakpoint = exports.BasicBox = exports.useBasicBox = exports.useBox = exports.Box = exports.useAvatar = exports.Avatar = exports.useCurls = exports.CurlsConsumer = exports.CurlsContext = exports.defaultColors = exports.defaultTheme = exports.ThemeProvider = exports.unit = exports.supportsCSS = exports.pushCss = exports.portalize = exports.nullIfFalsy = exports.memoThemeValue = exports.memoValue = exports.loadImage = exports.getHoverQuery = exports.getBreakpointOrder = exports.getDefault = exports.directionalScale = exports.colorize = exports.assignDefaults = exports.MAX_Z_INDEX = void 0

var _browserResets2 = _interopRequireDefault(require('./browserResets'))

exports.browserResets = _browserResets2.default

var _containmentAttrs2 = _interopRequireDefault(require('./containmentAttrs'))

exports.containmentAttrs = _containmentAttrs2.default

var _cloneElement2 = _interopRequireDefault(require('./cloneElement'))

exports.cloneElement = _cloneElement2.default

var _createComponent2 = _interopRequireDefault(require('./createComponent'))

exports.createComponent = _createComponent2.default

var _createRenderProp2 = _interopRequireDefault(require('./createRenderProp'))

exports.createRenderProp = _createRenderProp2.default

var _utils = require('./utils')

exports.assignDefaults = _utils.assignDefaults
exports.colorize = _utils.colorize
exports.directionalScale = _utils.directionalScale
exports.getDefault = _utils.get
exports.getBreakpointOrder = _utils.getBreakpointOrder
exports.getHoverQuery = _utils.getHoverQuery
exports.loadImage = _utils.loadImage
exports.memoValue = _utils.memoValue
exports.memoThemeValue = _utils.memoTheme
exports.nullIfFalsy = _utils.nullIfFalsy
exports.portalize = _utils.portalize
exports.pushCss = _utils.pushCss
exports.supportsCSS = _utils.supportsCSS
exports.unit = _utils.unit

var _ThemeProvider = require('./ThemeProvider')

exports.ThemeProvider = _ThemeProvider.ThemeProvider
exports.defaultTheme = _ThemeProvider.defaultTheme
exports.defaultColors = _ThemeProvider.defaultColors
exports.CurlsContext = _ThemeProvider.CurlsContext
exports.CurlsConsumer = _ThemeProvider.CurlsConsumer
exports.useCurls = _ThemeProvider.useCurls

var _Avatar = require('./Avatar')

exports.Avatar = _Avatar.Avatar
exports.useAvatar = _Avatar.useAvatar

var _Box = require('./Box')

exports.Box = _Box.Box
exports.useBox = _Box.useBox
exports.useBasicBox = _Box.useBasicBox
exports.BasicBox = _Box.BasicBox

var _Breakpoint = require('./Breakpoint')

exports.Breakpoint = _Breakpoint.Breakpoint
exports.useBreakpoint = _Breakpoint.useBreakpoint

var _Button = require('./Button')

exports.Button = _Button.Button
exports.useButton = _Button.useButton

var _Card = require('./Card')

exports.Card = _Card.Card
exports.useCard = _Card.useCard

var _Divider = require('./Divider')

exports.Divider = _Divider.Divider
exports.useDivider = _Divider.useDivider

var _Drawer = require('./Drawer')

exports.Drawer = _Drawer.Drawer
exports.DrawerBox = _Drawer.DrawerBox
exports.DrawerToggle = _Drawer.DrawerToggle
exports.useDrawerBox = _Drawer.useDrawerBox
exports.useDrawerContext = _Drawer.useDrawerContext
exports.DrawerContext = _Drawer.DrawerContext
exports.DrawerConsumer = _Drawer.DrawerConsumer

var _Drop = require('./Drop')

exports.Drop = _Drop.Drop
exports.useDrop = _Drop.useDrop

var _Fade = require('./Fade')

exports.Fade = _Fade.Fade
exports.useFade = _Fade.useFade

var _FlexGrid = require('./FlexGrid')

exports.FlexGrid = _FlexGrid.FlexGrid
exports.useFlexGrid = _FlexGrid.useFlexGrid

var _Flex = require('./Flex')

exports.Flex = _Flex.Flex
exports.useFlex = _Flex.useFlex

var _Grid = require('./Grid')

exports.Grid = _Grid.Grid
exports.useGrid = _Grid.useGrid

var _GridItem = require('./GridItem')

exports.GridItem = _GridItem.GridItem
exports.useGridItem = _GridItem.useGridItem

var _Input = require('./Input')

exports.Input = _Input.Input
exports.useInput = _Input.useInput

var _Link = require('./Link')

exports.Link = _Link.Link
exports.useLink = _Link.useLink
exports.A = _Link.A
exports.NavLink = _Link.NavLink

var _Modal = require('./Modal')

exports.Modal = _Modal.Modal
exports.ModalBox = _Modal.ModalBox
exports.ModalToggle = _Modal.ModalToggle
exports.useModalBox = _Modal.useModalBox
exports.useModalContext = _Modal.useModalContext
exports.ModalContext = _Modal.ModalContext
exports.ModalConsumer = _Modal.ModalConsumer

var _Popover = require('./Popover')

exports.Popover = _Popover.Popover
exports.PopoverMe = _Popover.PopoverMe
exports.PopoverBox = _Popover.PopoverBox
exports.usePopoverBox = _Popover.usePopoverBox
exports.usePopoverContext = _Popover.usePopoverContext
exports.PopoverContext = _Popover.PopoverContext
exports.PopoverConsumer = _Popover.PopoverConsumer

var _Slide = require('./Slide')

exports.Slide = _Slide.Slide
exports.useSlide = _Slide.useSlide

var _Text = require('./Text')

exports.Text = _Text.Text
exports.useText = _Text.useText
exports.prettyText = _Text.prettyText

var _TextArea = require('./TextArea')

exports.TextArea = _TextArea.TextArea
exports.useTextArea = _TextArea.useTextArea

var _Transitionable = require('./Transitionable')

exports.Transitionable = _Transitionable.Transitionable
exports.useTransitionable = _Transitionable.useTransitionable

var _createAriaPopup = require('./createAriaPopup')

exports.useAriaPopup = _createAriaPopup.useAriaPopup

var _createAriaPopupToggle = require('./createAriaPopupToggle')

exports.useAriaPopupToggle = _createAriaPopupToggle.useAriaPopupToggle

var _useBreakpointValueParser2 = _interopRequireDefault(
  require('./useBreakpointValueParser')
)

exports.useBreakpointValueParser = _useBreakpointValueParser2.default

var _useToggleVisibility2 = _interopRequireWildcard(
  require('./useToggleVisibility')
)

exports.useToggleVisibility = _useToggleVisibility2.default
exports.getDelay = _useToggleVisibility2.getDelay

var _useMediaQuery2 = _interopRequireDefault(require('./useMediaQuery'))

exports.useMediaQuery = _useMediaQuery2.default

var _core = require('@style-hooks/core')

Object.keys(_core).forEach(function(key) {
  if (key === 'default' || key === '__esModule') return
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return
  exports[key] = _core[key]
})

var _core2 = require('@emotion/core')

exports.css = _core2.css
exports.jsx = _core2.jsx
exports.withEmotionCache = _core2.withEmotionCache
exports.CacheProvider = _core2.CacheProvider
exports.Global = _core2.Global
exports.keyframes = _core2.keyframes
exports.ClassNames = _core2.ClassNames

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  } else {
    var newObj = {}
    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc =
            Object.defineProperty && Object.getOwnPropertyDescriptor
              ? Object.getOwnPropertyDescriptor(obj, key)
              : {}
          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc)
          } else {
            newObj[key] = obj[key]
          }
        }
      }
    }
    newObj.default = obj
    return newObj
  }
}

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {default: obj}
}

const MAX_Z_INDEX = 2147483647
exports.MAX_Z_INDEX = MAX_Z_INDEX
