// curls/dist/cjs/PropTypes.js
'use strict'

exports.__esModule = true
exports.boolOrNum = exports.boolOrString = exports.toggleVisibility = exports.positional = exports.implicitNum = exports.boolNumOrString = void 0

var _propTypes = require('prop-types')

const boolNumOrString = (0, _propTypes.oneOfType)([
  _propTypes.bool,
  _propTypes.number,
  _propTypes.string,
])
exports.boolNumOrString = boolNumOrString
const implicitNum = (0, _propTypes.oneOfType)([
  _propTypes.number,
  _propTypes.string,
  _propTypes.bool,
])
exports.implicitNum = implicitNum
const positional = {
  fromTop: boolNumOrString,
  fromRight: boolNumOrString,
  fromBottom: boolNumOrString,
  fromLeft: boolNumOrString,
}
exports.positional = positional
const toggleVisibility = {
  toggle: _propTypes.func,
  isVisible: _propTypes.bool,
  show: _propTypes.func,
  hide: _propTypes.func,
}
exports.toggleVisibility = toggleVisibility
const boolOrString = (0, _propTypes.oneOfType)([
  _propTypes.bool,
  _propTypes.string,
])
exports.boolOrString = boolOrString
const boolOrNum = (0, _propTypes.oneOfType)([
  _propTypes.bool,
  _propTypes.number,
  _propTypes.string,
])
exports.boolOrNum = boolOrNum
