// curls/dist/cjs/utils/index.js
'use strict'

exports.__esModule = true
exports.objectWithoutPropsMemo = exports.objectWithoutProps = exports.fastMemoize = exports.memoTheme = exports.memoValue = exports.isDirectional = void 0

var _assignDefaults2 = _interopRequireDefault(require('./assignDefaults'))

exports.assignDefaults = _assignDefaults2.default

var _colorize2 = _interopRequireDefault(require('./colorize'))

exports.colorize = _colorize2.default

var _directionalScale2 = _interopRequireWildcard(require('./directionalScale'))

exports.directionalScale = _directionalScale2.default
exports.isDirectional = _directionalScale2.isDirectional

var _get2 = _interopRequireDefault(require('./get'))

exports.get = _get2.default

var _getBreakpointOrder2 = _interopRequireDefault(
  require('./getBreakpointOrder')
)

exports.getBreakpointOrder = _getBreakpointOrder2.default

var _getHoverQuery2 = _interopRequireDefault(require('./getHoverQuery'))

exports.getHoverQuery = _getHoverQuery2.default

var _loadImage2 = _interopRequireDefault(require('./loadImage'))

exports.loadImage = _loadImage2.default

var _memoStyles = require('./memoStyles')

exports.memoValue = _memoStyles.memoValue
exports.memoTheme = _memoStyles.memoTheme

var _nullIfFalsy2 = _interopRequireDefault(require('./nullIfFalsy'))

exports.nullIfFalsy = _nullIfFalsy2.default

var _placeholder2 = _interopRequireDefault(require('./placeholder'))

exports.placeholder = _placeholder2.default

var _portalize2 = _interopRequireDefault(require('./portalize'))

exports.portalize = _portalize2.default

var _pushCss2 = _interopRequireDefault(require('./pushCss'))

exports.pushCss = _pushCss2.default

var _supportsCSS2 = _interopRequireDefault(require('./supportsCSS'))

exports.supportsCSS = _supportsCSS2.default

var _unit2 = _interopRequireDefault(require('./unit'))

exports.unit = _unit2.default

var _core = require('@style-hooks/core')

exports.fastMemoize = _core.fastMemoize
exports.objectWithoutProps = _core.objectWithoutProps
exports.objectWithoutPropsMemo = _core.objectWithoutPropsMemo

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  } else {
    var newObj = {}
    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc =
            Object.defineProperty && Object.getOwnPropertyDescriptor
              ? Object.getOwnPropertyDescriptor(obj, key)
              : {}
          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc)
          } else {
            newObj[key] = obj[key]
          }
        }
      }
    }
    newObj.default = obj
    return newObj
  }
}

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {default: obj}
}
