// curls/dist/cjs/Fade/styles.js
'use strict'

exports.__esModule = true
exports.isVisible = exports.isNotVisible_ = exports.isVisible_ = exports.baseIsVisible = exports.baseIsNotVisible = exports.to = exports.from = void 0

var _core = require('@emotion/core')

const from = () => null

exports.from = from

const to = () => null

exports.to = to
const baseIsNotVisible = {
  name: 'gyp8mm',
  styles: 'visibility:hidden;',
}
exports.baseIsNotVisible = baseIsNotVisible
const baseIsVisible = {
  name: 'rrm59m',
  styles: 'visibility:visible;',
}
exports.baseIsVisible = baseIsVisible
const isVisible_ =
  /*#__PURE__*/
  (0, _core.css)(baseIsVisible, ';opacity:1;')
exports.isVisible_ = isVisible_
const isNotVisible_ =
  /*#__PURE__*/
  (0, _core.css)(baseIsNotVisible, ';opacity:0;')
exports.isNotVisible_ = isNotVisible_

const isVisible = (value, _, props) =>
  value === true
    ? props.to === 1
      ? isVisible_
      : /*#__PURE__*/
        (0, _core.css)(baseIsVisible, ';opacity:', props.to, ';')
    : props.from === 0
    ? isNotVisible_
    : /*#__PURE__*/
      (0, _core.css)(baseIsNotVisible, ';opacity:', props.from, ';')

exports.isVisible = isVisible
